### DISCUSSION TITLE
Centroid only inference ?

### DISCUSSION THREAD

**Discussion by olivier-cuttlefish** [talmolab/sleap#2018] [Help!] [answered]
Hello, 

I would like to know if there would be an easy way to run inference using only centroid prediction. 
I have a lot of very long videos, and I am only interested in detecting the centroids of my instances. I would hope that only running the videos through the centroid models would save me a lot of processing time. I am expecting that going through Python would be the solution but I haven't managed to make anything work.

Thank you,

Olivier

**Comment by BenjaminBo** [comment, marked as answer]
Though I am not a developer for this repository, I am working with sleap alot in python at the moment and I have also tried to create a centroid-only model. 

Training a centroid model alone is possible (I think also actually the only way you can train it). 

When loading a model though you can't `sleap.load_model()` a centroid without an additional centered_instance model. It will throw you an error (that I can't check right now). 

What you could do is 
1. train a little dummy centered instance model that doesn't take much space and you would probably not even need to train it much (maybe only 1 epoch), 
2. load the centroid model you care about with the dummy centered instance model
3. run inference and do `labels = model.predict(data)` 
4. `labels` is an object that holds all your predictions. Extract the predictions you care about/ the centroid model made from `labels.predicted_instances`. Every instance has an attribute called `instance.bounding_box`.

Hope this helps :)

**Reply by olivier-cuttlefish** [reply]
Hi Benjamin, I haven't tried your method yet but on paper it does seems to be helpful indeed, thank you!

**Reply by talmo** [reply]
Thanks @BenjaminBo! That's indeed the only currently supported way to use centroid-only models, but we're hoping to change that at some point :)
